crates/ibc/src/core/ics03_connection/connection.rs

```diff
@@ -331,8 +331,12 @@ impl ConnectionEnd {
     }
 
     /// Getter for the list of versions in this connection end.
-    pub fn versions(&self) -> &[Version] {
-        &self.versions
+    pub fn versions(&self) -> Result<&[Version], ConnectionError> {
+        // Note: With the current implementation, only one version is supported per connection.
+        if self.versions.len() != 1 {
+            return Err(ConnectionError::InvalidVersionLength);
+        }
```

Comment:
I don't think ibc-go enforces this, right? If not, neither should we

Reply:
- The way a connection is created, neither us nor IBC-go will end up with a connection supporting multiple versions (`Vec<Version>`). Since [`get_compatible_versions`](https://github.com/cosmos/ibc-rs/blob/351efaeeda16621d2321832ecf953a00259067fe/crates/ibc/src/core/ics03_connection/version.rs#L117-L120) returns a vector of size 1, and the [`pick_version`](https://github.com/cosmos/ibc-go/blob/b07741cb95d3de9650564086ec3db55d1b854991/modules/core/03-connection/types/version.go#L146-L174) also matches a single version. 

- Though, already the `get_compatible_versions` API allows hosts to introduce multiple versions. Then, storing a `ConnectionEnd` on host with different versions is possible (bc of [here](https://github.com/cosmos/ibc-rs/blob/351efaeeda16621d2321832ecf953a00259067fe/crates/ibc/src/core/ics03_connection/handler/conn_open_init.rs#L35-L37)), which is not good looking at how a channel is initialized. The `MsgChannelOpenTry::validate()` method [doesn't know](https://github.com/cosmos/ibc-rs/blob/351efaeeda16621d2321832ecf953a00259067fe/crates/ibc/src/core/ics04_channel/handler/chan_open_init.rs#L28) which connection version should be checked for ordering.

I kept this condition as we are not supposed to see multiples upon calling `versions`. We had this before as `InvalidVersionLengthConnection` error. Perhaps, we should drop it after figuring out how to manage various versions or not allowing users for the `Vec<Version>` in `get_compatible_versions`. WDYT?

Reply:
See if this works? [63c50a3](https://github.com/cosmos/ibc-rs/pull/626/commits/63c50a34179e091703194dc9f17c516a53136360)